cd/publish_images.sh

```diff
@@ -16,27 +16,59 @@
 
 # Script for publishing to DockerHub
 # Requires the following environment variables to be set:
-# APP_VERSION - the release name
 # DOCKER_USERNAME - the dockerhub username
-# DOCKER_PASSWORD - the docker password
+# DOCKER_PASSWORD - the dockerhub password
+# GHCR_USERNAME - the ghcr.io username
+# GHCR_PASSWORD - the ghcr.io password/token
 
 # Gets project root directory by calling two nested "dirname" commands on the this file
 getRootDirectory() {
     echo "$( cd $(dirname $(dirname $0)) > /dev/null 2>&1 && pwd )"
 }
 
-# Pushes Tags to Dockerhub
-pushTags() {
+# Generate tags based on version
+getTags() {
+    if [[ $version =~ .*alpha.* ]];
+    then
+      tags="${name}:${version} ${name}:latest"
+    else
+      tags="$(echo ${version} | sed -E "s|([0-9]+)\.([0-9]+).*|${name}:${version} ${name}:\1.\2 ${name}:\1 ${name}:latest|")"
+    fi
+}
+
+# Generate tags for Gaffer image based on version
+getGafferTags() {
+    if [[ $version =~ .*alpha.* ]];
+    then
+      tags="${name}:${version} ${name}:latest"
+    else
+      tags="$(echo ${version} | sed -E "s|([0-9]+)\.([0-9]+)\.([0-9]+)-accumulo-([0-9.]*)|${name}:${version} ${name}:\1.\2.\3 ${name}:\1.\2 ${name}:\1 ${name}:latest|")"
```

Comment:
What does this mean? `${name}:\1.\2.\3`

Reply:
There's 3 capture groups, one for each part of the version. First group is Major, then Minor, then Patch. The other regex doesn't need this as the `$version` is already in this format and can be used directly.